week2/internal/controller/http/v1/actor_handler.go

```diff
@@ -79,3 +79,65 @@ func (handler *ActorHandler) Create(c *gin.Context) {
 	}
 	c.JSON(http.StatusOK, model.HttpResponse[entity.Actor]{Message: "Success", Data: actor})
 }
+
+// @Summary Update an actor
+// @Description Update an actor
+// @Tags Actor
+// @Accept json
+// @Param id path int true "actorId"
+// @RequestBody model.ActorRequest true "Actor payload"
+// @Produce  json
+// @Router /actors/{id} [put]
+// @Success 200 {object} entity.Actor
+func (handler *ActorHandler) Update(c *gin.Context) {
+	//check param id
+	id := c.Param("id")
+	parsedId, err := strconv.ParseInt(id, 10, 64)
+	if err != nil {
+
+		c.JSON(http.StatusBadRequest, model.HttpResponse[any]{Message: err.Error(), Data: nil})
+	}
+
+	var actorRequest model.ActorRequest
+	//binding request
+	if err1 := c.ShouldBindJSON(&actorRequest); err1 != nil {
+		c.JSON(http.StatusBadRequest, model.HttpResponse[any]{Message: err1.Error(), Data: nil})
+		return
+	}
+	//update
+	updatedActor := &entity.Actor{ID: parsedId, FirstName: actorRequest.FirstName, LastName: actorRequest.LastName}
```

Comment:
Please refer to the document about Update. I dont see they have to declare the ID in the update.
```
// Update attributes with `struct`, will only update non-zero fields
db.Model(&user).Updates(User{Name: "hello", Age: 18, Active: false})
// UPDATE users SET name='hello', age=18, updated_at = '2013-11-17 21:34:10' WHERE id = 111;
```

May be you can create an extra parameter in ```UpdateActor``` to receive an ```ID```. Like this
```actorService.UpdateActor(c.Request.Context(), updatedActor, parsedId)```. You can do this for the method in both service and repository

Reply:
i have received an id and put it into an entity.Actor:
id := c.Param("id")
	parsedId, err := strconv.ParseInt(id, 10, 64)
	if err != nil {

		c.JSON(http.StatusBadRequest, model.HttpResponse[any]{Message: err.Error(), Data: nil})
	}

	var actorRequest model.ActorRequest
	//binding request
	if err = c.ShouldBindJSON(&actorRequest); err != nil {
		c.JSON(http.StatusBadRequest, model.HttpResponse[any]{Message: err.Error(), Data: nil})
		return
	}
	//update
	updatedActor := &entity.Actor{ID: parsedId, FirstName: actorRequest.FirstName, LastName: actorRequest.LastName}

Reply:
Please follow the convention. Most Update function I see out there split the parameters in 2 parts, the ```ID``` of the resource and the ```payload```. Here you are combining the ```ID``` and the ```payload``` into that ```updatedUser```, which I dont recommend